lib/iolog/log_writer.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iolog
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"time"
+)
+
+const (
+	// MaxEntrySizeBytes is the maximum allowable size of a log entry
+	// before it will be broken into multiple entries
+	MaxEntrySizeBytes = 16 * 1024
```

Comment:
Is this 16K for a reason or is it just a random number we picked with a fair coin flip :) ?